virtualizarr/tests/test_xarray.py

```diff
@@ -419,6 +419,32 @@ def test_open_virtual_dataset_passes_expected_args(
         }
         mock_read_kerchunk.assert_called_once_with(**args)
 
+    @patch("virtualizarr.kerchunk.parse_array_refs")
```

Comment:
Wait what exactly is this mocking doing?

Reply:
I couldn't get a repro of the issue by just creating an xarray dataset with a 0-dim variable so I had to debug the problem I was getting on a real netCDF file and mock out the return value of `parse_array_refs` to match what I was seeing. Here's a repro if you'd like to give it shot:
```python
from urllib.parse import urlparse
import xarray as xr
import planetary_computer
import virtualizarr

def _href_to_fsspec(href: str) -> str:
    url = urlparse(href)
    return f"az://{url.path.lstrip('/')}"

def open_planetary_computer_netcdf(href):
    account_name = href.split("/")[2].split(".")[0]
    container_name = href.split("/")[3]
    fs = planetary_computer.get_adlfs_filesystem(account_name, container_name)
    return fs.open(_href_to_fsspec(href))

def _get_storage_options(href: str, use_sas_as_credential: bool = False) -> dict:
    url = urlparse(href)
    if use_sas_as_credential:
        return {
            "account_name": url.hostname.split(".")[0],
            "account_key": url.query,
            "anon": False,
        }
    else:
        return {
        "account_name": url.hostname.split(".")[0],
        "anon": False,
    }


href = "https://landcoverdata.blob.core.windows.net/esa-cci-lc/netcdf/C3S-LC-L4-LCCS-Map-300m-P1Y-2016-v2.1.1.nc"
# Next line works, you can inspect the `crs` variable
ds = xr.open_dataset(open_planetary_computer_netcdf(href))
# Next line will raise a `ValueError` unpacking a list of no elements when handling the `crs` kerchunk attrs
vds = virtualizarr.open_virtual_dataset(
    _href_to_fsspec(href),
    reader_options={
        "storage_options": _get_storage_options(planetary_computer.sign(href), use_sas_as_credential=True)
    })


```

Reply:
@ghidalgo3 I just met up with some folks from the HDF group at a conference last week so I can ask them the proper incantation to use so we can create an hdf fixture with a scalar 👍

Reply:
Thank you for the explanation! I'm happy to merge this as-is or wait if you wanted to test it some other way.